server/src/main/java/org/opensearch/index/mapper/BinaryFieldMapper.java

```diff
@@ -240,30 +241,28 @@ protected String contentType() {
      */
     public static class CustomBinaryDocValuesField extends CustomDocValuesField {
 
-        private final ObjectArrayList<byte[]> bytesList;
-
-        private int totalSize = 0;
+        private final ArrayList<byte[]> bytesList;
 
         public CustomBinaryDocValuesField(String name, byte[] bytes) {
             super(name);
-            bytesList = new ObjectArrayList<>();
+            bytesList = new ArrayList<>();
             add(bytes);
         }
 
         public void add(byte[] bytes) {
             bytesList.add(bytes);
-            totalSize += bytes.length;
         }
 
         @Override
         public BytesRef binaryValue() {
             try {
-                CollectionUtils.sortAndDedup(bytesList);
-                int size = bytesList.size();
-                BytesStreamOutput out = new BytesStreamOutput(totalSize + (size + 1) * 5);
-                out.writeVInt(size);  // write total number of values
-                for (int i = 0; i < size; i++) {
-                    final byte[] value = bytesList.get(i);
+                TreeSet<byte[]> bytesTree = new TreeSet((Comparator<byte[]>)((byte[] a, byte[] b) -> Arrays.compareUnsigned(a, b)));
+                bytesTree.addAll(bytesList);
+                int size = bytesTree.stream().mapToInt(b -> b.length).sum();
+                int length = bytesTree.size();
+                BytesStreamOutput out = new BytesStreamOutput(size + (length + 1) * 5);
```

Comment:
Minor, I was looking into this code before and we don't close the stream:

```
```
```suggestion
                try (BytesStreamOutput out = new BytesStreamOutput(size + (length + 1) * 5)) {
                }
```

 The `close()` is `noop` now but we should probably make sure the streams a closed anyway.

Reply:
Woo! Good catch!